### PR TITLE
Add Geolocation 

### LINKED ISSUE #1694: Remove geolocation from behind the feature flag and update the behaviors if necessary
**Is your feature request related to a problem? Please describe.**
Allow people to easily geolocate themselves on the map. Consider this as a default behavior for mobile.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by vim-usds**
# Purpose
- [x] Add Geolocation
- [x] closes #1694
- [x] add text as "Geolocating..." in black text with white background
- [x] change icon color to black
- [x] change beacon color to black
- [x] refactored most failed component due to USWDS upgrade
- ~still working on getting the beacon to turn off when off screen. One idea is to look for the `user.dot` class in the DOM during onClick and either remove or hide it, depending on what happens normally.~
- [x] still working on getting external links to have the proper vertical alignment
- [x] increase radius of beacon's inner border to X pixels
- [x] waiting on Kameron - new SVGs for waiting, active and background (if possible, checking contrast...)
- [x] add directional arrow during mobile [options.showUserHeading]
- [x] change disabled tool tip to "disabled"
- [x] get SSL on staging working
- [x] get app to switch between mobile and desktop views
- [x] test on mobile

 
### QA round 3 link [here](https://screeningtool-staging.geoplatform.gov/1746-71c78d/en/#3/33.47/-97.5)

QA spec (desktop)
- [x] When you click on the Geolocate button, does it show you your location?
- [ ] is the button's styling correct?
- [x] is the beacon's styling correct?
- [ ] is the message during 'geolocating" correct?

QA spec (mobile)
- [x] Does the geolocation work on mobile?
- [x] does it track you while on mobile?

### CODE REVIEW

File: client/src/components/J40Map.module.scss [hunk 1]
```diff
@@ -1,36 +1,75 @@
...
 
-.fullscreenControl {
-  right: 1.25em;
-  top: 2.5em;
-}
+  .mapHeaderRow{
+    @include u-display("flex");
+    @include u-padding-left(1.5);
+    
+    justify-content: space-between;
```
Comment (TomNUSDS) [inline client/src/components/J40Map.module.scss:15]: I cannot find `justify-content: space-between` in the [caniuse site](https://caniuse.com/?search=justify-content). Can you verify it's supported by all your target browsers?

File: client/src/components/J40Map.tsx [hunk 2]
```diff
@@ -346,9 +351,10 @@ const J40Map = ({location}: IJ40Interface) => {
          *   - npm run clean && npm run build && npm run serve
          *
          * to ensure the production build works and that MapSearch and the map (ReactMapGL) render correctly.
+         *
+         * Any component declrations outside the <ReactMapGL> component may be susceptible to this bug.
```
Comment (TomNUSDS) [inline client/src/components/J40Map.tsx:355]: typo `declarations`

File: client/src/components/J40Map.tsx [hunk 3]
```diff
@@ -346,9 +351,10 @@ const J40Map = ({location}: IJ40Interface) => {
...
          */}
-        <MapSearch goToPlace={goToPlace}/>
-
+        {/* <MapSearch goToPlace={goToPlace}/> */}
```
Comment (TomNUSDS) [inline client/src/components/J40Map.tsx:357]: Maybe just remove?

**Comment by katherinedm-usds** [comment]
Worked on personal Macbook w/ Chrome
image

**Review by TomNUSDS — APPROVED** [review]
LGTM. Maybe verify CSS feature?

**Comment by KameronKerger** [comment]
it did not work on safari (desktop), chrome (desktop), or chrome (mobile) for me. I got the circle can't use on all three.
Screen Shot 2022-07-14 at 1 58 51 PM
maybe I was too late? @vim-usds ?

**Comment by vim-usds** [comment]
> it did not work on safari (desktop), chrome (desktop), or chrome (mobile) for me. I got the circle can't use on all three. Screen Shot 2022-07-14 at 1 58 51 PM maybe I was too late? @vim-usds ?

@KameronKerger - I have a feeling we can get it work maybe. Let me know if you have some time to zoom together. I can drop a link

File: client/src/components/J40Map.module.scss (hunk 1 above)
Comment (vim-usds) [reply client/src/components/J40Map.module.scss:15]: Great catch! Works on Chrome 103 , FF 102 and Safari 15.5

**Comment by KameronKerger** [comment]
there are still a couple of things
if you are geolocating and you press the button you get stuck in this nothing is happening state?
Screen Shot 2022-07-19 at 11 03 56 AM
same thing if you are geolocating in another area and you try to turn it off
Screen Shot 2022-07-19 at 11 06 03 AM
do we have to have the blue outline on the icon?
I think the states may need to be updated to: off, in process (geolocating), on - centered, on-not centered

**Comment by KameronKerger** [comment]
@vim-usds 
icon default = leave as is
beacon = 0076d6 with a 4 pt stroke
icon waiting = 0076d6 spinning
icon active = 0076d6
icon panned = 0076d6

**Comment by vim-usds** [comment]
@KameronKerger - Thank you!

**Comment by vim-usds** [comment]
@KameronKerger 

## QA round 3 link [here](http://usds-geoplatform-justice40-website.s3-website-us-east-1.amazonaws.com/justice40-tool/1746-a1a047/en/#3/33.47/-97.5)

If the icon is disabled, please see the instructions at the start of this thread.

### QA spec (desktop)

- [ ] When you click on the Geolocate button, does it show you your location?
- [ ] is the button's styling correct?
- [ ] is the beacon's styling correct?
- [ ] is the message during 'geolocating" correct?

**Comment by KameronKerger** [comment]
the behavior on desktop still seems a little off / funky. if you press the button after you are located it gets stuck in the "locating" step and there's no way to turn it off.
Screen Shot 2022-08-01 at 10 02 10 AM
I'd also like to move the button on mobile and desktop as shown in these two screenshots (move closer to the search bar and up on mobile)
Screen Shot 2022-08-01 at 10 02 10 AM_move
mobile
I'll have to test the following moving on mobile next time I take odin out for a walk.

**Comment by KameronKerger** [comment]
mobile tracking is good. it doesn't have direction though... is that easy to add? supported by Mapbox?

**Comment by vim-usds** [comment]
@KameronKerger  - I think we will have to be on a newer version of MapBox to get that directional feature on mobile. Also I'm unable to recreate the "2nd-click-on-icon-makes-it-freeze" issue. Maybe we can do a quick screen share today.

**Pull request closed without merging**